### PR TITLE
Update to std-future

### LINKED ISSUE #38: Update to std-futures
Now that hyper's `master` branch supports `std::future`, it would be very useful to have `hyper-tls` support `std::future` as well.

### PR DESCRIPTION
**Pull request by 95th**
Closes #38.

Uses tokio-tls

### CODE REVIEW

**Comment by sfackler** [comment]
It seems like this should just wrap tokio-tls to avoid re-implementing all of that stuff.

**Comment by jakubadamw** [comment]
I've a got a WIP branch here: https://github.com/jakubadamw/hyper-tls/tree/tokio-tls. It is, however, blocked on a [small change](https://github.com/tokio-rs/tokio/pull/1224/files#r304272254) to `tokio-tls` that would make `TlsStream` a `Send` type again, as it was accidentally made non-`Send` during the move to `std::futures`.

**Comment by jakubadamw** [comment]
`TlsStream` will be made `Sync` again after https://github.com/tokio-rs/tokio/pull/1317.

@95th, if you don't mind, when that's merged in, I'll pick up my branch, cherry-pick your fixes to examples and create a PR. OK?

**Comment by jakubadamw** [comment]
@95th, I rebased my branch against yours but I got stuck on just one remaining error in my changes. It seems I need to read up on `Pin` and `Unpin` before I can solve it so if you feel like picking up my branch and taking a look, I will not mind either!

This is what I'm getting:

```
error[E0599]: no method named `poll` found for type `std::pin::Pin<&mut std::boxed::Box<(dyn core::future::future::Future<Output = std::result::Result<(stream::MaybeHttpsStream<T>, hyper::client::connect::Connected), std::io::Error>> + std::marker::Send + 'static)>>` in the current scope
   --> src\client.rs:134:31
    |
134 |         Pin::new(&mut self.0).poll(cx)
    |                               ^^^^
    |
    = note: the method `poll` exists but the following trait bounds were not satisfied:
            `std::boxed::Box<(dyn core::future::future::Future<Output = std::result::Result<(stream::MaybeHttpsStream<T>, hyper::client::connect::Connected), std::io::Error>> + std::marker::Send + 'static)> : core::future::future::Future`
```

Adding an `Unpin` bound to `BoxFut` doesn't quite help as then it complains about the `GenFuture` returned by `TlsConnector::connect()` (as that's an `async fn`) not being `Unpin`.

**Comment by 95th** [comment]
@jakubadamw  I fixed that issue and made a PR to your repo. Do you want to create a separate PR or do you want to merge your changes in this PR itself?

**Comment by jakubadamw** [comment]
@95th, great, thanks! Please go ahead and just carry all over to this PR. :)

**Comment by 95th** [comment]
@jakubadamw Done. Thanks

**Comment by jakubadamw** [comment]
Oh, the `scoped-tls` dependency is no longer needed. I was going to remove it in my commit but then it slipped my mind. :)

File: Cargo.toml [hunk 1]
```diff
@@ -8,16 +8,18 @@ keywords = ["hyper", "tls", "http", "https", "ssl"]
...
-bytes = "0.4"
-futures = "0.1.21"
+futures-preview = { version = "=0.3.0-alpha.17", features = ["async-await", "nightly"] }
 native-tls = "0.2"
-hyper = "0.12"
-tokio-io = "0.1"
+hyper = { git = "https://github.com/hyperium/hyper" }
+tokio-io = { git = "https://github.com/tokio-rs/tokio" }
+tokio-tls = { git = "https://github.com/tokio-rs/tokio" }
+scoped-tls = "1.0"
```
Comment (jakubadamw) [inline Cargo.toml]: The scoped-tls dependency is no longer needed. I was going to remove it in my commit but then it slipped my mind. :)

Comment (95th) [reply Cargo.toml]: gone. Thanks.

File: src/stream.rs [hunk 2]
```diff
@@ -56,129 +37,50 @@ impl<T> From<TlsStream<T>> for MaybeHttpsStream<T> {
...
-
-// ===== impl TlsStream =====
-
-impl<T> TlsStream<T> {
-    pub(crate) fn new(inner: native_tls::TlsStream<T>) -> Self {
-        TlsStream {
-            inner,
-        }
+    fn poll_flush(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Result<(), io::Error>> {
+        Poll::Ready(Ok(()))
```
Comment (whataloadofwhat) [inline src/stream.rs]: Is there a reason that this doesn't defer its implementation to the inner writer like the other methods?

Comment (95th) [reply src/stream.rs]: Missed by mistake. Fixed now. Thanks!

File: Cargo.toml [hunk 3]
```diff
@@ -8,16 +8,17 @@ keywords = ["hyper", "tls", "http", "https", "ssl"]
...
 documentation = "https://docs.rs/hyper-tls"
+edition = "2018"
 
 [features]
 vendored = ["native-tls/vendored"]
 
 [dependencies]
-bytes = "0.4"
-futures = "0.1.21"
+futures-preview = { version = "=0.3.0-alpha.17", features = ["async-await", "nightly"] }
```
Comment (seanmonstar) [inline Cargo.toml]: If we're already using `async`/`await`, then we don't even need this dependency, right? The `future::err` can just be `async { Err(..) }`.

Comment (95th) [reply Cargo.toml]: You're right. Updated. PTAL.

**Pull request merged**